x-pack/legacy/plugins/apm/public/components/app/ServiceMap/index.tsx

```diff
@@ -37,37 +54,159 @@ ${theme.euiColorLightShade}`,
   margin: `-${theme.gutterTypes.gutterLarge}`
 };
 
+const MAX_REQUESTS = 5;
+
 export function ServiceMap({ serviceName }: ServiceMapProps) {
-  const {
-    urlParams: { start, end }
-  } = useUrlParams();
+  const callApmApi = useCallApmApi();
+  const license = useLicense();
+  const { search } = useLocation();
+  const { urlParams, uiFilters } = useUrlParams();
+  const { notifications } = useApmPluginContext().core;
+  const params = useDeepObjectIdentity({
+    start: urlParams.start,
+    end: urlParams.end,
+    environment: urlParams.environment,
+    serviceName,
+    uiFilters: {
+      ...uiFilters,
+      environment: undefined
+    }
+  });
```

Comment:
I'm not following why this is necessary.  If we need `environment` on the backend we can get it from `uiFilters.environment`. No need to introduce this complexity by rewriting it. 

Not sure this is related but the environment selector currently doesn't work. When changing environment the service map is unaffected and it's necessary to hard refresh to see the correct service map. Wouldn't